From 8d00cd5d2f9f707c7959f9d7b825e6b6e7f523f2 Mon Sep 17 00:00:00 2001
From: Takuto Ikuta <tikuta@chromium.org>
Date: Wed, 18 Jan 2023 02:45:22 +0000
Subject: [PATCH] grit: remove dependency to six

Bug: 1406153
Change-Id: I2cf74b2885b640c9639971d509038e08c2c5d157
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4173416
Auto-Submit: Takuto Ikuta <tikuta@chromium.org>
Commit-Queue: Andrew Grieve <agrieve@chromium.org>
Reviewed-by: Andrew Grieve <agrieve@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1093706}
---
Upstream-Status: Backport [8dca1dabe7759924d853fb63cd138768fc61de47 http://gpro.lge.com/c/webosose/chromium108/+/389012 grit: remove dependency to six]

 src/tools/binary_size/supersize.pydeps      | 1 -
 src/tools/grit/BUILD.gn                     | 1 -
 src/tools/grit/grit/__init__.py             | 8 --------
 src/tools/grit/grit/clique.py               | 2 --
 src/tools/grit/grit/grd_reader.py           | 2 --
 src/tools/grit/grit/tclib.py                | 2 --
 src/tools/grit/grit/tool/android2grd.py     | 1 -
 src/tools/grit/grit/tool/build.py           | 2 --
 src/tools/grit/grit/tool/menu_from_parts.py | 2 --
 src/tools/grit/grit/tool/rc2grd.py          | 1 -
 src/tools/grit/grit/tool/xmb.py             | 2 --
 src/tools/grit/grit/util_unittest.py        | 2 --
 src/tools/grit/pak_util.py                  | 6 ------
 src/tools/grit/setup.py                     | 4 ----
 14 files changed, 36 deletions(-)

diff --git a/src/tools/binary_size/supersize.pydeps b/src/tools/binary_size/supersize.pydeps
index b4e2533898..e0ae3e91dd 100644
--- a/src/tools/binary_size/supersize.pydeps
+++ b/src/tools/binary_size/supersize.pydeps
@@ -1,6 +1,5 @@
 # Generated by running:
 #   build/print_python_deps.py --root tools/binary_size --output tools/binary_size/supersize.pydeps tools/binary_size/libsupersize/main.py
-../../third_party/six/src/six.py
 ../grit/grit/__init__.py
 ../grit/grit/clique.py
 ../grit/grit/constants.py
diff --git a/src/tools/grit/BUILD.gn b/src/tools/grit/BUILD.gn
index dfcfbe8289..49cdbd99f4 100644
--- a/src/tools/grit/BUILD.gn
+++ b/src/tools/grit/BUILD.gn
@@ -35,7 +35,6 @@ group("grit_python_unittests") {
     "//tools/grit/",
     "//third_party/catapult/third_party/typ/",
     "//third_party/node/",
-    "//third_party/six/src/six.py",
   ]
 }
 
diff --git a/src/tools/grit/grit/__init__.py b/src/tools/grit/grit/__init__.py
index fa82d3a86e..5ff2ce636c 100644
--- a/src/tools/grit/grit/__init__.py
+++ b/src/tools/grit/grit/__init__.py
@@ -5,11 +5,3 @@
 '''Package 'grit'
 '''
 
-
-import os
-import sys
-
-
-_HERE_PATH = os.path.dirname(__file__)
-_SRC_PATH = os.path.normpath(os.path.join(_HERE_PATH, '..', '..', '..'))
-sys.path.insert(0, os.path.join(_SRC_PATH, 'third_party', 'six', 'src'))
diff --git a/src/tools/grit/grit/clique.py b/src/tools/grit/grit/clique.py
index 01560f6e44..714b9e7078 100644
--- a/src/tools/grit/grit/clique.py
+++ b/src/tools/grit/grit/clique.py
@@ -9,8 +9,6 @@ collections of cliques (uber-cliques).
 
 import re
 
-import six
-
 from grit import constants
 from grit import exception
 from grit import lazy_re
diff --git a/src/tools/grit/grit/grd_reader.py b/src/tools/grit/grit/grd_reader.py
index 5cc3702d89..99ed30f12c 100755
--- a/src/tools/grit/grit/grd_reader.py
+++ b/src/tools/grit/grit/grd_reader.py
@@ -12,8 +12,6 @@ import sys
 import xml.sax
 import xml.sax.handler
 
-import six
-
 from grit import exception
 from grit import util
 from grit.node import mapping
diff --git a/src/tools/grit/grit/tclib.py b/src/tools/grit/grit/tclib.py
index 9e991c635a..1bd0c57ce6 100644
--- a/src/tools/grit/grit/tclib.py
+++ b/src/tools/grit/grit/tclib.py
@@ -9,8 +9,6 @@
 import functools
 import re
 
-import six
-
 from grit import exception
 from grit import lazy_re
 import grit.extern.tclib
diff --git a/src/tools/grit/grit/tool/android2grd.py b/src/tools/grit/grit/tool/android2grd.py
index 8526a7ef28..908ae691ed 100644
--- a/src/tools/grit/grit/tool/android2grd.py
+++ b/src/tools/grit/grit/tool/android2grd.py
@@ -11,7 +11,6 @@ import sys
 from xml.dom import Node
 import xml.dom.minidom
 
-import six
 from io import StringIO
 
 import grit.node.empty
diff --git a/src/tools/grit/grit/tool/build.py b/src/tools/grit/grit/tool/build.py
index 664b9f6e6d..f3f61c4cf2 100644
--- a/src/tools/grit/grit/tool/build.py
+++ b/src/tools/grit/grit/tool/build.py
@@ -15,8 +15,6 @@ import os
 import shutil
 import sys
 
-import six
-
 from grit import grd_reader
 from grit import shortcuts
 from grit import util
diff --git a/src/tools/grit/grit/tool/menu_from_parts.py b/src/tools/grit/grit/tool/menu_from_parts.py
index 7e0466a85e..075edc8d4c 100644
--- a/src/tools/grit/grit/tool/menu_from_parts.py
+++ b/src/tools/grit/grit/tool/menu_from_parts.py
@@ -5,8 +5,6 @@
 '''The 'grit menufromparts' tool.'''
 
 
-import six
-
 from grit import grd_reader
 from grit import util
 from grit import xtb_reader
diff --git a/src/tools/grit/grit/tool/rc2grd.py b/src/tools/grit/grit/tool/rc2grd.py
index 291f2e71bd..e0282d9760 100644
--- a/src/tools/grit/grit/tool/rc2grd.py
+++ b/src/tools/grit/grit/tool/rc2grd.py
@@ -10,7 +10,6 @@ import getopt
 import re
 import sys
 
-import six
 from io import StringIO
 
 import grit.node.empty
diff --git a/src/tools/grit/grit/tool/xmb.py b/src/tools/grit/grit/tool/xmb.py
index 1c27d8ac33..646bdd9238 100644
--- a/src/tools/grit/grit/tool/xmb.py
+++ b/src/tools/grit/grit/tool/xmb.py
@@ -12,8 +12,6 @@ import sys
 
 from xml.sax import saxutils
 
-import six
-
 from grit import grd_reader
 from grit import lazy_re
 from grit import tclib
diff --git a/src/tools/grit/grit/util_unittest.py b/src/tools/grit/grit/util_unittest.py
index 342d7c238e..71640dfc1f 100755
--- a/src/tools/grit/grit/util_unittest.py
+++ b/src/tools/grit/grit/util_unittest.py
@@ -14,8 +14,6 @@ if __name__ == '__main__':
 
 import unittest
 
-import six
-
 from grit import util
 
 
diff --git a/src/tools/grit/pak_util.py b/src/tools/grit/pak_util.py
index 4a0b5dc18e..49f16220fe 100755
--- a/src/tools/grit/pak_util.py
+++ b/src/tools/grit/pak_util.py
@@ -19,12 +19,6 @@ import subprocess
 import sys
 import tempfile
 
-_HERE_PATH = os.path.dirname(__file__)
-_SRC_PATH = os.path.normpath(os.path.join(_HERE_PATH, '..', '..'))
-sys.path.insert(0, os.path.join(_SRC_PATH, 'third_party', 'six', 'src'))
-
-import six
-
 from grit import constants
 from grit.format import data_pack
 
diff --git a/src/tools/grit/setup.py b/src/tools/grit/setup.py
index cc2d80afa0..3784add67c 100755
--- a/src/tools/grit/setup.py
+++ b/src/tools/grit/setup.py
@@ -16,9 +16,6 @@ setuptools.setup(
         'console_scripts': ['grit = grit.grit_runner:Main'],
     },
     packages=setuptools.find_packages(),
-    install_requires=[
-        'six >= 1.10',
-    ],
     author='The Chromium Authors',
     author_email='chromium-dev@chromium.org',
     description=('Google Resource and Internationalization Tool for managing '
@@ -34,7 +31,6 @@ setuptools.setup(
         'Operating System :: Microsoft :: Windows',
         'Operating System :: POSIX :: Linux',
         'Programming Language :: Python',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
